### DISCUSSION TITLE
Typesafe enum query param in RPC client?

### DISCUSSION THREAD

**Discussion by branislav-brincko** [honojs/hono#2475] [General]
This is my backend: 

![image](https://github.com/honojs/hono/assets/118365024/fa8f6652-24d2-47ec-918a-3cbcdbd7c65f)

And this is my frontend: 

![image](https://github.com/honojs/hono/assets/118365024/4355660c-8ad5-46d7-9206-1d74ae90aff2)

Is there a way how to make "order" on the frontend inferred not as string | string[] but as a "asc" | "desc"? (In other words, how to make typescript check if I passed the correct value?)

**Comment by yusukebe** [comment]
Hi @branislav-brincko 

This is not a bug but rather an unimplemented one. It can be a feature request.

**Comment by askorupskyy** [comment]
You could still have it in your types by using `z.union([z.literal("asc"), z.literal("desc")])`

**Reply by askorupskyy** [reply]
It makes even more sense when thought of in terms of TypeScript.
```typescript
type Order = 'asc' | 'desc';
```
TS enums suck 😁

**Reply by branislav-brincko** [reply]
This still does not provide correct typing on frontend ⬇️ `order` is typed as `string | string[]` and I can pass anything to it without warning. 

![image](https://github.com/honojs/hono/assets/118365024/cb54bd5f-f04a-496e-b9f7-07e7e44e5cdc)

**Reply by askorupskyy** [reply]
Yeah this is definitely not implemented then. I'll try to play around and see if i can get it to work
